persistence/sql/persister_client.go

```diff
@@ -73,15 +80,16 @@ func (p *Persister) DeleteClient(ctx context.Context, id string) error {
 		return err
 	}
 
-	return sqlcon.HandleError(p.Connection(ctx).Destroy(&client.Client{ID: cl.ID}))
+	return sqlcon.HandleError(p.Connection(ctx).Destroy(&client.Client{ID: cl.ID, NID: p.NetworkID(ctx)}))
```

Comment:
Destroy only uses the primary key, so the NID doesn't have an effect here. You will need to use RawQUERY instead